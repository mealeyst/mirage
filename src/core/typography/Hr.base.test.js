import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import BaseHr from './Hr.base'

const { shallowWithTheme } = global

describe('(Styled Component) BaseHr', () => {
  const createBaseHr = (props) => {
    return shallowWithTheme(<BaseHr {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBaseHr())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '80%'
    expect(createBaseHr({width: width}))
    .toHaveStyleRule({
      width: width
    })
  })
})
