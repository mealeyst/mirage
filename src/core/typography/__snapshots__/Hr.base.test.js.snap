// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) BaseHr matching the snapshot 1`] = `
.c0 {
  height: .1rem;
  width: 100%;
  border: none;
  margin: 1rem 0;
}

<UnstyledHr
  className="c0"
  width="100%"
/>
`;
